custom_components/xiaomi_miio_fan/fan.py

```diff
@@ -442,10 +447,15 @@ def supported_features(self) -> int:
             self._state_attrs.update(
                 {key: self._extract_value_from_attribute(state, value) for
                  key, value in self._available_attributes.items()})
+            self._retry = 0
 
         except DeviceException as ex:
-            self._available = False
-            _LOGGER.error("Got exception while fetching the state: %s", ex)
+            self._retry = self._retry + 1
+            if self._retry < self._retries:
+                _LOGGER.warning("Got exception while fetching the state: %s , _retry=%s", ex, self._retry)
+            else:
+                self._available = False
+                _LOGGER.error("Got exception while fetching the state: %s , _retry=%s", ex, self._retry)
```

Comment:
The log message is equal in both cases. Please invert the if condition and move the log message above.

Reply:
Ping! :-)